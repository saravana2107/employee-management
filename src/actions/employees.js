import {
  CREATE_EMPLOYEE,
  DELETE_EMPLOYEE,
  TOGGLE_EMPLOYEE_MODAL
} from "../constants/ActionTypes";
import { toast } from "react-toastify";

/**
 * Employee Create
 *
 * @param payload
 * @param isEdit
 * @returns {{type, payload: *}}
 */
export function add(payload, isEdit = false) {
  toast.success(`Employee ${isEdit ? "Updated" : "Created"} Successfully`, {
    hideProgressBar: false
  });
  return { type: CREATE_EMPLOYEE, payload };
}

/**
 * Employee Delete
 *
 * @param payload
 * @returns {{type, payload: *}}
 */
export function del(payload) {
  toast.success("Employee Deleted Successfully", {
    hideProgressBar: false
  });
  return { type: DELETE_EMPLOYEE, payload };
}

/**
 * Toggle Modal
 *
 * @param payload
 * @returns {{type, payload: *}}
 */
export function toggleModal(payload = null) {
  return { type: TOGGLE_EMPLOYEE_MODAL, payload };
}
