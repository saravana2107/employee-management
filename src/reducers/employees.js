import {
  CREATE_EMPLOYEE,
  DELETE_EMPLOYEE,
  TOGGLE_EMPLOYEE_MODAL
} from "../constants/ActionTypes";

import _ from "lodash";

export function employees(
  state = {
    modal: false,
    data: {},
    employeeId: null
  },
  action
) {
  switch (action.type) {
    case CREATE_EMPLOYEE:
      return Object.assign({}, state, {
        modal: false,
        data: Object.assign({}, state.data, {
          [action.payload.id]: action.payload
        })
      });
    case DELETE_EMPLOYEE:
      return Object.assign({}, state, {
        modal: false,
        data: _.omit(state.data, action.payload)
      });
    case TOGGLE_EMPLOYEE_MODAL:
      return Object.assign({}, state, {
        modal: !state.modal,
        employeeId: action.payload
      });
    default:
      return state;
  }
}
