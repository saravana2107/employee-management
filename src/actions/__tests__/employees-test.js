import { add, del, toggleModal } from "../employees";

describe("employees actions", () => {
  it("should add employee", () => {
    expect(
      add({
        id: "ALTI123456",
        name: "Test User #1",
        email: "test+1@altimetric.com",
        phone: "9876543210",
        address: "Kudlu Gate",
        city: "Bangalore",
        state: "Karnataka",
        zipcode: "560068",
        gender: "Male"
      })
    ).toMatchSnapshot();
  });

  it("should delete employee", () => {
    expect(del("ALTI123456")).toMatchSnapshot();
  });

  it("should toggle modal", () => {
    expect(toggleModal()).toMatchSnapshot();
  });

  it("should toggle modal with employee id", () => {
    expect(toggleModal("ALTI123456")).toMatchSnapshot();
  });
});
