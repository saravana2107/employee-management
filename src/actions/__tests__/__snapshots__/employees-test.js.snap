// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`employees actions should add employee 1`] = `
Object {
  "payload": Object {
    "address": "Kudlu Gate",
    "city": "Bangalore",
    "email": "test+1@altimetric.com",
    "gender": "Male",
    "id": "ALTI123456",
    "name": "Test User #1",
    "phone": "9876543210",
    "state": "Karnataka",
    "zipcode": "560068",
  },
  "type": "CREATE_EMPLOYEE",
}
`;

exports[`employees actions should delete employee 1`] = `
Object {
  "payload": "ALTI123456",
  "type": "DELETE_EMPLOYEE",
}
`;

exports[`employees actions should toggle modal 1`] = `
Object {
  "payload": null,
  "type": "TOGGLE_EMPLOYEE_MODAL",
}
`;

exports[`employees actions should toggle modal with employee id 1`] = `
Object {
  "payload": "ALTI123456",
  "type": "TOGGLE_EMPLOYEE_MODAL",
}
`;
